src/test/java/org/sagebionetworks/EvaluationUtilsTest.java

```diff
@@ -74,9 +90,149 @@ public void testUpdateSubmissionRemove() throws Exception {
+	public void testSetStatusCancelRequest() throws Exception {
+		SubmissionStatusModifications statusMods = new SubmissionStatusModifications();
+		assertNotNull(statusMods.getAnnotationsToAdd());
+		assertEquals(0, statusMods.getAnnotationsToAdd().size());
+		SubmissionStatusEnum submissionStatusEnum = SubmissionStatusEnum.INVALID;
+		WorkflowUpdateStatus containerStatus = WorkflowUpdateStatus.STOPPED_UPON_REQUEST;
+		setStatus(statusMods, submissionStatusEnum, containerStatus);
+		assertNotNull(statusMods);
+		assertEquals(submissionStatusEnum, statusMods.getStatus());
+		assertNotNull(statusMods.getAnnotationsToAdd());
+		assertEquals(1, statusMods.getAnnotationsToAdd().size());
+		String expectedKey = "orgSagebionetworksSynapseWorkflowOrchestratorStatusDescription";
+		assertNotNull(expectedKey, statusMods.getAnnotationsToAdd().get(0).getKey());
+	}
+
+	@Test
+	public void testApplyModificationsString() throws Exception {
+		SubmissionStatusModifications statusMods = new SubmissionStatusModifications();
+		SubmissionStatus expected = new SubmissionStatus();
+		SubmissionStatus actual = new SubmissionStatus();
+
+		setAnnotation(expected, "foo1", "bar", false);
+		setAnnotationStringV2(expected, "foo1", "bar", false);
+		setAnnotation(statusMods, "foo1", "bar", false);
+
+		// Call under test
+		applyModifications(actual, statusMods);
+		assertEquals(expected, actual);
+	}
+
+	@Test
+	public void testApplyModificationsLong() throws Exception {
+		SubmissionStatusModifications statusMods = new SubmissionStatusModifications();
+		SubmissionStatus expected = new SubmissionStatus();
+		SubmissionStatus actual = new SubmissionStatus();
+
+		setAnnotation(expected, "foo2", 1L, false);
+		setAnnotationLongV2(expected, "foo2", 1L, false);
+		setAnnotation(statusMods, "foo2", 1L, false);
+
+		// Call under test
+		applyModifications(actual, statusMods);
+		assertEquals(expected, actual);
+	}
+
+	@Test
+	public void testApplyModificationsDouble() throws Exception {
+		SubmissionStatusModifications statusMods = new SubmissionStatusModifications();
+		SubmissionStatus expected = new SubmissionStatus();
+		SubmissionStatus actual = new SubmissionStatus();
+
+		setAnnotation(expected, "foo3", 3.14D, true);
+		setAnnotationDoubleV2(expected, "foo3", 3.14D, false);
+		setAnnotation(statusMods, "foo3", 3.14D, true);
+
+		// Call under test
+		applyModifications(actual, statusMods);
+		assertEquals(expected, actual);
+	}
+
+	@Test
+	public void testRemoveAnnotationsStringV2() throws Exception {
+		SubmissionStatus actual = new SubmissionStatus();
+		setAnnotationStringV2(actual, "foo1", "baz1", false);
+		// Call under test
+		removeAnnotation(actual, "foo1");
+		assertNotNull(actual);
+		assertNotNull(actual.getSubmissionAnnotations());
+		assertNotNull(actual.getSubmissionAnnotations().getAnnotations());
+		assertEquals(0, actual.getSubmissionAnnotations().getAnnotations().size());
+	}
+
+	@Test
+	public void testRemoveAnnotationsLongV2() throws Exception {
```

Comment:
Testing for different types is a bit out of scope since the remove acts on a key. On the other end you could have a test with multiple annotations with different types, making sure to get the expected result (e.g. that only the key you are passing has been removed from the map).